### DISCUSSION TITLE
Lossless Video

### DISCUSSION THREAD

**Discussion by BrittJuliff** [raspberrypi/picamera2#628] [Q&A]
What is the best way to output a lossless video with Picamera2? 

Thanks!

**Comment by davidplowman** [comment]
There is no specific support for any lossless video codecs. I suppose there are a few options:

* Use the `Encoder` base class rather than the `H264Encoder` or anything else. Using the base class will cause uncompressed video frames to be saved. Obviously these will be large and you may be limited by the speed at which the output can be written. It will depend to a significant extent on the resolution and size of your frames. YUV420 will be considerably better than RGB888, though depending on your interpretation, you might not regard that as strictly "lossless".

* If you are a slightly flexible about your definition of "lossless", then saving h.264 or MJPEG at very high bitrates may be "close enough"?

**Reply by BrittJuliff** [reply]
Thanks!
